ui/app/scripts/shared/services/parser.js

```diff
@@ -700,13 +705,44 @@ define(function() {
 	        	}
         		node.errors.push(error);
         	}
+			
+			function eatTaskDefinition(lineNum) {
+				var taskNode = {};
+				var taskName = maybeEatStreamName();
```

Comment:
This function `maybeEatStreamName()` name can probably be generic as there is an error message which says `Illegal Stream Name` for an illegal task name here.